vanilla/vanilla
# Add threshold checking to MySQL index modifications

Vanilla's alter table thresholds are not currently respected for index or "additional" queries being run. Since these operations can also have performance penalties, we're implementing threshold checking to these changes as well. All MySQL table modification queries should now be covered.

In addition, we're implementing default options for index operations: [inplace](https://dev.mysql.com/doc/refman/5.6/en/alter-table.html#alter-table-performance) and [no-lock](https://dev.mysql.com/doc/refman/5.6/en/alter-table.html#alter-table-concurrency). If the index operation fails, another attempt will be made without these options.

### Testing
1. Set your `Database.AlterTableThreshold` config to something small, like 1.
1. Delete an index from your discussions table.
1. Run a utility-update. You should see an error indicating your index could not be recreated due to the alter table threshold being exceeded.
1. Delete the `Database.AlterTableThreshold` config.
1. Run another utility-update. Your index should be created without error.
